content/docs/visualization/template_examples.md

```diff
@@ -0,0 +1,113 @@
+[Go templating](http://golang.org/pkg/text/template/) system.
+
+## Simple alert field templates
+
+    ALERT InstanceDown
+      IF up == 0
+      FOR 5m
+      WITH {
+        severity="page"
+      }
+      SUMMARY "Instance {{$labels.instance}} down"
+      DESCRIPTION "{{$labels.instance}} of job {{$labels.job}} has been down for more than 5 minutes."
+
+
+Alert field templates will be executed during every rule iteration for each
+alert that fires, so keep any queries and templates lightweight. If you have a
+need for more complicated templates for alerts, it is recommended to link to a
+console instead.
+
+## Simple iteration
+
+This displays a list of hosts, and whether they are up:
+
+```
+{{ range query "up" }}
+  {{ .Labels.instance }} {{ .Labels.Value }}
+{{ end }}
+```
+
+The special `.` variable contains the value of the current sample for each loop iteration.
+
+## Display one value
+
+```
+{{ with query "some_metric{instance='someinstance'}" }}
+{{ . | first | value | humanize }}
+{{ end }}
+```
+
+Go and Go's templating language are both strongly typed, so one must check that
+samples were returned to avoid an execution error. For example this could
+happen if a scrape or rule evaluation has not run yet, or a host was down.
+
+The included `prom_query_drilldown` template handles this, allows for
+formatting of results and linking to the [expression browser](../browser/).
+
+## Using console URL parameters
+
+```
+{{ with printf "node_memory_MemTotal{job='node',instance='%s'}" .Params.instance | query }}
+{{ . | first | value | humanize1024}}B
+{{ end }}
+```
+
+If accessed as `console.html?instance=hostname`, `.Params.instance` will evaluate to `hostname`.
+
+## Advanced iteration
+
+```html
+<table>
+{{ range printf "node_network_receive_bytes{job='node',instance='%s',device!='lo'}" .Params.instance | query | sortByLabel "device"}}
+  <tr><th colspan=2>{{ .Labels.device }}</th></tr>
+  <tr>
+    <td>Received</td>
+    <td>{{ with printf "rate(node_network_receive_bytes{job='node',instance='%s',device='%s'}[5m])" .Labels.instance .Labels.device | query }}{{ . | first | value | humanize }}B/s{{end}}</td>
+  </tr>
+  <tr>
+    <td>Transmitted</td>
+    <td>{{ with printf "rate(node_network_transmit_bytes{job='node',instance='%s',device='%s'}[5m])" .Labels.instance .Labels.device | query }}{{ . | first | value | humanize }}B/s{{end}}</td>
+  </tr>{{ end }}
+<table>
+```
+
+Here we iterate over the network devices, then for each of them display the network traffic. 
+
+As the range doesn't specify a variable, `.Params.instance` isn't available
+inside the loop as `.` is now the loop variable.
+
+## Basic templates
+
+There's support for defining templates that can be reused. This is particularly
+powerful when combined with [console library](../template_reference/#console-templates)
+support, allowing sharing of templates across consoles.
+
+```
+{{define "myTemplate"}}
+  do something
+{{end}}
+{{template "myTemplate"}}
```

Comment:
Done, but with go template comments (console_libraries can only contain whitespace and template definitions, html comments would break things if someone did a simple copy&paste).